src/dtemplate.d

```diff
@@ -4059,27 +4062,46 @@ extern (C++) MATCH deduceType(RootObject o, Scope* sc, Type tparam, TemplatePara
             {
                 result = m;
             }
+            else if (e.type.toBasetype().isintegral() ||
```

Comment:
What about static arrays?  Can't this be checked using type converts to tailconst(type)?